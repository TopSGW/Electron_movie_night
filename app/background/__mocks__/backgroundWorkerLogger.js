'use strict'
/* globals jest */

// Swallow all log messages during testing
const logger = {
  debug: jest.fn(),
  info: jest.fn(),
  warn: jest.fn(),
  error: jest.fn()
}

export default logger
