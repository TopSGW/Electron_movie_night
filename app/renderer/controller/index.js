// Logic for mapping application state onto component props.
import {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  SearchMovies,
  MainContent,
  MovieThumbnailContainer
} from './containers'

// Logic for mapping application events directly to redux dispatch actions.
import './mapEventsToDispatch'

// Re-export all the containers.
export {
  App,
  DisplayMoviesContainer,
  ImportMovies,
  ImportStats,
  MainContent,
  MovieThumbnailContainer,
  SearchMovies
}
