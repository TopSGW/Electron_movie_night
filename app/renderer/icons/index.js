import cn from 'classnames'
import React from 'react'

const Icon = ({ className, border, fixedWidth, large, pulse, size3x, ...props }) => {
  return <i
    className={cn(
      'fa',
      className,
      pulse && 'fa-pulse',
      large && 'fa-lg',
      fixedWidth && 'fa-fw',
      size3x && 'fa-3x',
      border && 'fa-border'
    )}
    {...props}
  />
}

export const Anchor = ({ className, ...props }) => {
  return <Icon className={cn('fa-anchor', className)} {...props} />
}

export const Angle = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-angle-down',
      left && 'fa-angle-left',
      right && 'fa-angle-right',
      up && 'fa-angle-up',
      !down && !left && !right && !up && 'fa-angle-right', // default
      className
    )}
    {...props}
  />
}

export const Bars = ({ className, ...props }) => {
  return <Icon className={cn('fa-bars', className)} {...props} />
}

export const Ban = ({ className, ...props }) => {
  return <Icon className={cn('fa-ban', className)} {...props} />
}

export const Chevron = ({ className, down, left, right, up, ...props }) => {
  return <Icon
    className={cn(
      down && 'fa-chevron-down',
      left && 'fa-chevron-left',
      right && 'fa-chevron-right',
      up && 'fa-chevron-up',
      !down && !left && !right && !up && 'fa-chevron-right', // default
      className
    )}
    {...props}
  />
}

export const Database = ({ className, ...props }) => {
  return <Icon className={cn('fa-database', className)} {...props} />
}

export const Close = ({ className, ...props }) => {
  return <Icon className={cn('fa-close', className)} {...props} />
}

export const Edit = ({ className, ...props }) => {
  return <Icon className={cn('fa-pencil', className)} {...props} />
}

export const FileIcon = ({ className, video, ...props }) => {
  return <Icon
    className={cn(
      video && 'fa-file-video-o',
      !video && 'fa-file', // default
      className
    )}
    {...props}
  />
}

export const ImportIcon = ({ className, ...props }) => {
  return <Icon
    className={cn(
      'fa-download',
      className
    )}
    {...props}
  />
}

export const Info = ({ className, circle, ...props }) => {
  return <Icon
    className={cn(
      circle && 'fa-info-circle',
      !circle && 'fa-info',
      className
    )}
    {...props}
  />
}

export const Play = ({ className, ...props }) => {
  return <Icon className={cn('fa-play', className)} {...props} />
}

export const Search = ({ className, ...props }) => {
  return <Icon className={cn('fa-search', className)} {...props} />
}

export const Spinner = ({ className, ...props }) => {
  return <Icon className={cn('fa-spinner', className)} {...props} />
}

export const Star = ({ className, half, outline, ...props }) => {
  return <Icon
    className={cn(
      half && 'fa-star-half-o',
      outline && 'fa-star-o',
      !half && !outline && 'fa-star', // default
      className
    )}
    {...props}
  />
}

export const Trash = ({ className, ...props }) => {
  return <Icon className={cn('fa-trash', className)} {...props} />
}
