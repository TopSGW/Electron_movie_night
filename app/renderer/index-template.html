<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <title>Movie Night</title>
    <link rel="stylesheet" href="../vendor/font-awesome-4.7.0/css/font-awesome.min.css">
    <%=htmlWebpackPlugin.options.css%>
  </head>
  <body style="background-color:rgb(20,20,20) !important; /* override bootstrap */">
    <div id="root"></div>
    <script src="<%=htmlWebpackPlugin.options.bundle%>"></script>
  </body>
</html>
