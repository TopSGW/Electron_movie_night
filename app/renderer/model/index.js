import { ipcRenderer } from 'electron'
import { createStore, applyMiddleware, combineReducers } from 'redux'
import reduxLogger from 'redux-logger'

import { MOVE_MOVIE_TO_TRASH, UPDATE_MOVIE_METADATA } from '../../shared/events'
import { isDevEnv } from '../../shared/utils'
import appLogger from '../mainWindowLogger'

// Reducers
import dbLoaded from './dbLoaded'
import crawl from './crawl'
import featuredMovie from './featuredMovie'
import movies from './movies'
import search from './search'

// Top level state selectors.
import {
  crawlActiveSelector,
  crawlStatsSelector,
  showCrawlStatsSelector,
  movieDBSelector,
  searchCategorySelector,
  searchQuerySelector
} from './selectors'

// Memoized Selectors
import enhancedFeaturedMovieSelector from './enhancedFeaturedMovie'
import visibleMoviesSelector from './visibleMovies'

// Re-export all the action creators.
// ----------------------------------
export * from './actions'

// Electron action creators.
// -------------------------
export const updateMetadataFor = (movie) => {
  ipcRenderer.send(UPDATE_MOVIE_METADATA, movie)
  appLogger.info('Sent UPDATE_MOVIE_METADATA event', { movie: movie.title })
}

export const deleteMovie = (movie) => {
  ipcRenderer.send(MOVE_MOVIE_TO_TRASH, movie)
  appLogger.info(`Sent ${MOVE_MOVIE_TO_TRASH} event`, { movie: movie.title })
}

// Public selectors.
// These allow connected containers to depend on state
// without requiring knowledge of the state's shape.
// ----------------------------------------------------------------------
export const getAllMovies = (state) => {
  return movieDBSelector(state)
}

export const getCrawlActive = (state) => {
  return crawlActiveSelector(state)
}

export const getCrawlStats = (state) => {
  return crawlStatsSelector(state)
}

export const getShowCrawlStats = (state) => {
  return showCrawlStatsSelector(state)
}

export const getFeaturedMovie = (state) => {
  return enhancedFeaturedMovieSelector(state)
}

export const getSearchCategory = (state) => {
  return searchCategorySelector(state)
}

export const getSearchQuery = (state) => {
  return searchQuerySelector(state)
}

export const getVisibleMovies = (state) => {
  return visibleMoviesSelector(state)
}

// Export function for creating redux store.
// -----------------------------------------
export const createReduxStore = () => {
  // Create the redux store.
  // This is the default export.
  // ---------------------------
  let middlewares = []
  if (isDevEnv()) {
    middlewares.push(reduxLogger)
  }

  const rootReducer = combineReducers({
    crawl,
    dbLoaded,
    featuredMovie,
    movies,
    search
  })

  const store = createStore(rootReducer, applyMiddleware(...middlewares))
  return store
}

// This is the actual store used in the app.
// -----------------------------------------
const store = createReduxStore()
export default store
