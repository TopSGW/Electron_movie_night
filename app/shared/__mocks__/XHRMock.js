'use strict'

class XHRMockBase {
  constructor () {
    this.timeout = 0
    this.responseType = null
    this.status = null
    this.statusText = null
    this.response = null
    this.mock = jest.fn() /* globals jest */
  }
  open (...args) { this.mock(...args) }
}

class XHRMockSuccess extends XHRMockBase {
  send () {
    this.status = 200
    this.statusText = 'OK'
    this.response = this.mock.mock
    this.onload()
  }
}

class XHRMock404NotFound extends XHRMockBase {
  send () {
    this.status = 404
    this.statusText = 'Not Found'
    this.onload()
  }
}

class XHRMockNetworkError extends XHRMockBase {
  send () {
    this.onerror()
  }
}

class XHRMockTimeoutError extends XHRMockBase {
  send () {
    this.ontimeout()
  }
}

const jsonTestInput = '{"test": ["a","b","c",1,2,3]}'
const jsonTestOutput = JSON.parse(jsonTestInput)
class XHRMockJSON extends XHRMockBase {
  send () {
    this.status = 200
    this.statusText = 'OK'
    this.response = jsonTestInput
    this.onload()
  }
}

class XHRMockGetFirstSuccess extends XHRMockBase {
  send () {
    const url = this.mock.mock.calls[0][1] // from call to req.open()
    this.status = 200
    this.statusText = 'OK'
    this.response = JSON.stringify(url)
    this.onload()
  }
}

module.exports = {
  XHRMockSuccess,
  XHRMock404NotFound,
  XHRMockNetworkError,
  XHRMockTimeoutError,
  XHRMockJSON,
  jsonTestOutput,
  XHRMockGetFirstSuccess
}
