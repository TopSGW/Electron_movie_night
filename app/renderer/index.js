import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'

import { logEnv } from '../shared/utils'
import { App } from './controller'
import store from './model'
import logger from './mainWindowLogger'

logEnv(logger)

// Render the App.
ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
)
