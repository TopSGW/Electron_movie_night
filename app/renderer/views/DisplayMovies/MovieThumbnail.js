import { shell } from 'electron'
import React, { Component } from 'react'
import styled from 'styled-components'
import { fileExists } from '../../../shared/utils'
import { Angle } from '../../icons'
import { fadeIn, fadeOut } from '../styleUtils'
import PlayButton from './MovieDetail/PlayMovieButton'

export default class MovieThumbnail extends Component {
  constructor (props) {
    super(props)
    this.state = {
      controls: 'hide', // hide, fadingIn, fadingOut, show
      image: 'show', // show, fadeOut
      fileAvailable: true
    }

    this.mouseEnter = this.mouseEnter.bind(this)
    this.mouseLeave = this.mouseLeave.bind(this)
    this.onAnimationEnd = this.onAnimationEnd.bind(this)

    this.openMovieInDefaultPlayer = this.openMovieInDefaultPlayer.bind(this)
    this.onShowMovieDetailsClick = this.onShowMovieDetailsClick.bind(this)
    this.renderControls = this.renderControls.bind(this)
  }

  mouseEnter (e) {
    e.preventDefault()
    this.setState({ controls: 'fadingIn' })
    fileExists(this.props.movie.location)
      .then(result => this.setState({ fileAvailable: result }))
  }

  mouseLeave (e) {
    e.preventDefault()
    if (this.state.controls === 'show') {
      this.setState({ controls: 'fadingOut' })
    } else {
      this.setState({ controls: 'hide' })
    }
  }

  onAnimationEnd (e) {
    e.preventDefault()
    if (this.state.controls === 'fadingIn') {
      this.setState({ controls: 'show' })
    } else if (this.state.controls === 'fadingOut') {
      this.setState({ controls: 'hide' })
    }
  }

  openMovieInDefaultPlayer (e) {
    e.preventDefault()
    shell.openItem(this.props.movie.location)
  }

  onShowMovieDetailsClick (e) {
    e.preventDefault()
    const { movie, handleMovieSelected, panelID } = this.props
    if (handleMovieSelected) {
      handleMovieSelected({ movie, action: 'click', panelID })
    }
  }

  render () {
    return (
      <ImageContainer
        onMouseEnter={this.mouseEnter}
        onMouseLeave={this.mouseLeave}
      >
        {this.renderImage()}
        {this.renderControls()}
      </ImageContainer>
    )
  }

  renderImage () {
    const { imgFile, title } = this.props.movie

    if (!imgFile) {
      return <NoImage>{title}</NoImage>
    }

    return <Image
      src={imgFile}
      alt={title}
    />
  }

  renderControls () {
    if (this.state.controls === 'hide') {
      return null
    }

    const Tag = this.state.controls === 'fadingOut' ? HoverOut : HoverIn

    return (
      <Tag onAnimationEnd={this.onAnimationEnd} >
        {this.state.fileAvailable && <PlayButton onClick={this.openMovieInDefaultPlayer} small />}
        <ShowMovieDetailsArrow
          down
          onClick={this.onShowMovieDetailsClick}
        />
      </Tag>
    )
  }
}

const ImageContainer = styled.div`
  margin: 5px 10px 5px 0;
  background-color: #141414;
  color: #999;
  display: flex;
  position: relative;
`

const Image = styled.img`
  width: 150px;
  height: 222px;
`

const NoImage = styled.div`
  width: 150px;
  height: 222px;
  border: 1px solid #999;
  padding: 8px;
  overflow-wrap: break-word;
`

const Hover = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`

const HoverIn = Hover.extend`
  animation: 0.5s ${fadeIn} ease-in;
`

const HoverOut = Hover.extend`
  animation: 0.3s ${fadeOut} ease-out;
`

const ShowMovieDetailsArrow = styled(Angle)`
  color: rgba(255, 255, 255, 0.9);
  cursor: pointer;
  position: absolute;
  bottom: 0;
  font-size: 50px;
  margin-bottom: -5px;
  overflow: hidden;
  text-align: center;
  text-shadow: 0 0 1px #141414;
  transition: all 0.2s ease-in-out;
  &:hover {
    color: rgba(2, 117, 216, 1);
  }
`
