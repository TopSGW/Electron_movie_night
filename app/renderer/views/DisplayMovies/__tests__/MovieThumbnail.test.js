'use strict'
/* globals jest, describe, test, expect */

import React from 'react'
import renderer from 'react-test-renderer'

import { shellMock } from '../../../../../__mocks__/electron'
import MovieThumbnail from '../MovieThumbnail'

jest.mock('../../../../shared/utils', () => {
  return {
    fileExists: () => Promise.resolve(true)
  }
})

// { movie, handleMovieSelected, panelID } = this.props

const movie = {
  actors: ['Luke Wilson', 'Will Ferrell'],
  director: 'Todd Phillips',
  imgFile: '/path/to/poster',
  location: '/path/to/movie',
  genres: ['action', 'comedy'],
  title: 'Old School'
}

const handleSelection = jest.fn()

describe('MovieThumbnail', () => {
  test('renders', () => {
    const component = renderer.create(
      <MovieThumbnail
        movie={movie}
        panelID={1}
        handleMovieSelected={handleSelection}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Trigger a mouseenter event.
    // This will cause the play button and selection arrow to
    // appear on next render.
    tree.props.onMouseEnter({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Trigger animation end render controls.
    let controls = tree.children[1]
    controls.props.onAnimationEnd({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Make sure click on play button calls Electron's shell API
    const playButton = controls.children[0].children[0]
    playButton.props.onClick({ preventDefault: jest.fn() })
    expect(shellMock).toHaveBeenLastCalledWith(movie.location)

    // Make sure click on details arrow calls the handleSelection callback
    const detailsArrow = controls.children[1]
    detailsArrow.props.onClick({ preventDefault: jest.fn() })
    expect(handleSelection).toHaveBeenLastCalledWith({
      movie,
      action: 'click',
      panelID: 1
    })

    // Trigger a mouseleave event.
    // The play button and selection arrows should disappear
    // on next render followiwng animation end event.
    tree.props.onMouseLeave({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    controls = tree.children[1]
    controls.props.onAnimationEnd({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    // Trigger a mouseenter and then mouseleave with no corresponding
    // animation end in between. This should result in the controls not
    // being rendered because fadeOut will be skipped.
    tree.props.onMouseEnter({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()

    tree.props.onMouseLeave({ preventDefault: jest.fn() })
    tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
