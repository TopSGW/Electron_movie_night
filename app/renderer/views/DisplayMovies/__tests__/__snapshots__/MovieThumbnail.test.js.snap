// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumbnail renders 1`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
</div>
`;

exports[`MovieThumbnail renders 2`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
  <div
    className="sc-bZQynM ixKulh"
    onAnimationEnd={[Function]}
  >
    <div
      className="sc-bdVaJa bqeGKh"
      onClick={undefined}
    >
      <i
        className="fa fa-play sc-bwzfXH fnOVYt"
        onClick={[Function]}
      />
    </div>
    <i
      className="fa fa-angle-down sc-htoDjs mwGmO"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`MovieThumbnail renders 3`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
  <div
    className="sc-bZQynM ixKulh"
    onAnimationEnd={[Function]}
  >
    <div
      className="sc-bdVaJa bqeGKh"
      onClick={undefined}
    >
      <i
        className="fa fa-play sc-bwzfXH fnOVYt"
        onClick={[Function]}
      />
    </div>
    <i
      className="fa fa-angle-down sc-htoDjs mwGmO"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`MovieThumbnail renders 4`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
  <div
    className="sc-gzVnrw bkbqMW"
    onAnimationEnd={[Function]}
  >
    <div
      className="sc-bdVaJa bqeGKh"
      onClick={undefined}
    >
      <i
        className="fa fa-play sc-bwzfXH fnOVYt"
        onClick={[Function]}
      />
    </div>
    <i
      className="fa fa-angle-down sc-htoDjs mwGmO"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`MovieThumbnail renders 5`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
</div>
`;

exports[`MovieThumbnail renders 6`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
  <div
    className="sc-bZQynM ixKulh"
    onAnimationEnd={[Function]}
  >
    <div
      className="sc-bdVaJa bqeGKh"
      onClick={undefined}
    >
      <i
        className="fa fa-play sc-bwzfXH fnOVYt"
        onClick={[Function]}
      />
    </div>
    <i
      className="fa fa-angle-down sc-htoDjs mwGmO"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`MovieThumbnail renders 7`] = `
<div
  className="sc-htpNat hBSyaA"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    alt="Old School"
    className="sc-bxivhb jTaKmc"
    src="/path/to/poster"
  />
</div>
`;
